### DISCUSSION TITLE
Capstone - Tidak Bisa Menjalankan Flask

### DISCUSSION THREAD

**Discussion by DGUM27** [teamalgoritma/community#199] [Q&A] [answered]
Dear All, 

Mohon bantuan, saya mengalami kendala dalam menginput flask di dalam visual code untuk capstone webscrapping. terima kasih

![2022-07-11 10_04_29-app py - Capstone - Dwi - Visual Studio Code  Administrator](https://user-images.githubusercontent.com/107378736/178181257-65ce014d-1582-4c5e-b879-6dc372de3f77.png)

![2022-07-11 10_07_22-Gagal Import Flask (Capstone API) · Discussion #178 · teamalgoritma_community](https://user-images.githubusercontent.com/107378736/178181403-ead246ee-0b64-4ef5-8f0a-0ac1aff07983.png)

**Comment by triarm** [comment, marked as answer]
Hi @DGUM27 ,
error tersebut terjadi karena Anda belum mengubah interpreter Python ke environment capstonenya. Berikut adalah cara untuk mengganti python interpreter:

1. Pada Visual Studio Code tekan Ctrl+Shift+P untuk menampilkan fitur search
2. Silahkan ketikan "python interpreter"
3. Pilih "Python: Select Interpreter"
4. Pilih interpreter python sesuai dengan environment yang telah dibuat

Demikian yang dapat kami sampaikan. Semoga menjawab

**Reply by DGUM27** [reply]
Dear Triam, 

Terima kasih banyak atas feedback yang diberikan untuk error case saya. 
Setelah menjalankan instruksi dari Triam, berikut muncul notifikasi seperti ini. Please your help, Triam. 
thanks in advance.


![2022-07-11 10_23_33-](https://user-images.githubusercontent.com/107378736/178182753-22e41bd5-33ae-43ef-a341-a92c2c472442.png)

**Reply by triarm** [reply]
Hi @DGUM27 ,
Error tersebut terjadi karena terminal yang digunakan adalah Powershell bukan Command Prompt. Untuk mengganti terminal menjadi Command Prompt, Anda dapat melihat solusi dari [diskusi berikut](https://github.com/teamalgoritma/community/discussions/180)

Semoga menjawab

**Reply by DGUM27** [reply]
Hi @triarm thanks untuk error case tsb sudah solved.

Mohon bantuan dari Triam untuk solve kendala di dalam menjalankan vscode sebagaimana terlampir. Terima kasih banyak.

![1](https://user-images.githubusercontent.com/107378736/178190920-06502109-c1cc-4b07-b7a3-0f5a551e5277.png)
![2](https://user-images.githubusercontent.com/107378736/178190927-b9313232-22d2-455b-9053-6a30c868b5b7.png)
![3](https://user-images.githubusercontent.com/107378736/178190941-e44d693d-a8fb-4c03-9df3-8ac47ed954fe.png)
![4](https://user-images.githubusercontent.com/107378736/178191012-73512230-47cf-4bc2-b468-be661c21b390.png)

**Reply by triarm** [reply]
Hi @DGUM27 ,
error tersebut terjadi karena Anda mencoba menghitung mean dari kolom Date saat mendefinisikan object card_data. Silahkan Anda untuk mengganti kolom tersebut menjadi kolom yang lebih sesuai untuk dihitung nilai rata-ratanya.

Semoga menjawab :)

**Reply by DGUM27** [reply]
Hi @triarm thanks for your advice and help untuk case tersebut sudah closed juga. 

Saya ada kendala lagi di part ini, sebagaimana saya lampirkan vscode di bawah ini. 
Mohon bantuan @triarm perihal ini. Thanks in advance.

![1](https://user-images.githubusercontent.com/107378736/178240635-3d8309d3-ce1f-420a-a673-6cd15e78413a.png)

![2](https://user-images.githubusercontent.com/107378736/178240705-8b8ec434-bcef-492b-91c3-948d04d168b1.png)

![3](https://user-images.githubusercontent.com/107378736/178240720-8cb07bfa-275e-439a-a580-b5486fd1ca9c.png)

![4](https://user-images.githubusercontent.com/107378736/178240739-76bf2831-1f35-4c7a-9131-98162bcb9997.png)

**Reply by triarm** [reply]
Error tersebut terjadi karena tidak ditemukannya file index.html pada direktori Anda. Untuk mengatasi hal tersebut, Anda perlu memastikan apakah struktur penyimpanan filenya sudah sesuai dengan struktur yang ada pada GitLab repository yang sudah disediakan dengan folder templates berisikan file index.html dan folder static berisikan file style.css
![image](https://user-images.githubusercontent.com/20942472/178252252-e8d0b663-0502-4955-b2c5-4ce2f66cfa39.png)

Semoga membantu

**Reply by DGUM27** [reply]
@triarm thanks for helping me.

Ketika saya coba running mengapa tidak muncul grafik sebagaimana gambar terlampir, serta bagaimana cara untuk menampilkannya?

![1](https://user-images.githubusercontent.com/107378736/178324122-c475cc22-3bfd-47d5-b888-ca9f123a29ca.png)

![2](https://user-images.githubusercontent.com/107378736/178324132-55c4e829-f09e-4fc6-b599-7600bb29c528.png)

![3](https://user-images.githubusercontent.com/107378736/178324155-068661fe-e1f3-4d45-924e-165eee1f58ba.png)

![4](https://user-images.githubusercontent.com/107378736/178324164-887b8ecb-53a3-488d-b25b-9855f0ca1873.png)

**Reply by triarm** [reply]
Hi @DGUM27 ,
Pada proses looping
```
for i in range(1, len('tr')):
   ...
```
menunjukkan bahwa looping tersebut akan berulang sepanjang string `"tr"` yaitu dua kali. Jika maksud Anda adalah menginginkan proses looping berulang sepanjang hasil pencarian tag `"tr"` silahkan ganti `"tr"` menjadi `tr`

Semoga membantu

**Reply by DGUM27** [reply]
Hi @triarm thank you so much for helping me fixing my error cases. I really appreciate it.

**Reply by triarm** [reply]
Hi @DGUM27 ,
you're welcome. Good luck for your capstone!
